plugins/modules/identity/keycloak/keycloak_identity_provider.py

```diff
@@ -0,0 +1,576 @@
+        default: 'present'
+        type: str
+        choices:
+            - present
+            - absent
+
+    realm:
+        description:
+            - The Keycloak realm under which the identity provider resides.
+        default: 'master'
+        type: str
+
+    alias:
+        description:
+            - The alias uniquely identifies an identity provider and it is also used to build the redirect uri.
+            - This parameter is required.
+        required: true
+        type: str
+
+    display_name:
+        description:
+            - Friendly name for identity provider.
+        type: str
+
+    enabled:
+        description:
+            - Enable/disable this identity provider.
+        type: bool
+
+    store_token:
+        description:
+            - Enable/disable if tokens must be stored after authenticating users.
+        aliases:
+            - storeToken
+        type: bool
+
+    add_read_token_role_on_create:
+        description:
+            - Enable/disable if new users can read any stored tokens. This assigns the broker.read-token role.
+        aliases:
+            - addReadTokenRoleOnCreate
+        type: bool
+
+    trust_email:
+        description:
+            - If enabled, email provided by this provider is not verified even if verification is enabled for the realm.
+        aliases:
+            - trustEmail
+        type: bool
+
+    link_only:
+        description:
+            - If true, users cannot log in through this provider. They can only link to this provider.
+              This is useful if you don't want to allow login from the provider, but want to integrate with a provider.
+        aliases:
+            - linkOnly
+        type: bool
+
+    hide_on_login_page:
+        description:
+            - If hidden, login with this provider is possible only if requested explicitly, for example using the 'kc_idp_hint' parameter.
+        aliases:
+            - hideOnLoginPage
+        type: bool
+
+    gui_order:
+        description:
+            - Number defining order of the provider in GUI (for example, on Login page).
+        aliases:
+            - guiOrder
+        type: int
+
+    first_broker_login_flow_alias:
+        description:
+            - Alias of authentication flow, which is triggered after first login with this identity provider.
+        aliases:
+            - firstBrokerLoginFlowAlias
+        type: str
+
+    post_broker_login_flow_alias:
+        description:
+            - Alias of authentication flow, which is triggered after each login with this identity provider.
+        aliases:
+            - postBrokerLoginFlowAlias
+        type: str
+
+    sync_mode:
+        description:
+            - Default sync mode for all mappers. The sync mode determines when user data will be synced using the mappers.
+        aliases:
+            - syncMode
+        type: str
+
+    provider_id:
+        description:
+            - Protocol used by this provider (oidc, saml, ...).
```

Comment:
Assuming you are listing possible values for this option:
```suggestion
            - Protocol used by this provider (C(oidc), C(saml), ...).
```
If you mean the protocols, you should use the correct casing (OIDC, SAML).

Reply:
Yes, these are possible values for the parameter